### PR TITLE
Add support for audio feedback if placing node failed

### PR DESCRIPTION
**Pull request by BlockMen**
As requested in https://github.com/minetest/minetest/issues/1128

### CODE REVIEW

**Comment by est31** [comment]
It isn't needed to raise the protocol version for this.

**Comment by est31** [comment]
This only plays back if the prediction failed, not the actual placement, right?

**Comment by BlockMen** [comment]
> It isn't needed to raise the protocol version for this.

Oh, right.

> This only plays back if the prediction failed, not the actual placement, right?

Well, yes, serverside handling is not touched.

**Comment by PilzAdam** [comment]
@est31 if placement fails on the server, then it's always caused in Lua, right? That would mean mods / games that want this could play the sound themselves then.

**Comment by est31** [comment]
That's an argument.

**Comment by BlockMen** [comment]
Removed protocol bump

**Comment by est31** [comment]
Looks good.

File: src/game.cpp [hunk 1]
```diff
@@ -3677,6 +3677,9 @@ void Game::handlePointingAtNode(GameRunData *runData,
 				if (playeritem_def.node_placement_prediction == "" ||
 						nodedef_manager->get(map.getNodeNoEx(nodepos)).rightclickable)
 					client->interact(3, pointed); // Report to server
+				else
+					soundmaker->m_player_rightpunch_sound =
+						playeritem_def.sound_place_failed;
```
Comment (kwolekr) [inline src/game.cpp]: Oh jeez, please add brackets around the bodies of this if..else statement, please, I feel it would help clarify things a lot.

**Comment by kwolekr** [comment]
Looks good.

File: src/game.cpp (hunk 1 above)
Comment (BlockMen) [inline src/game.cpp]: Personally i prefer the bracket-less solution, but fine, will change it.

**Comment by celeron55** [comment]
Currently this sound is used in only a few special cases: 1) the mapblock where the node is being placed isn't loaded, 2) the node being pointed is not buildable_to, 3) the node being placed is attached_node and the pointed node is not walkable, 4) the node is being placed where the player is standing.

Anyway, +1.

**Pull request merged**